pkg/password/password_policies.go

```diff
@@ -0,0 +1,167 @@
+package password
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+	"strings"
+	"unicode/utf8"
+)
+
+var defaultSpecialCharacters = "!\"#$%&'()*+,-./:;<=>?@[\\]^_`{|}~"
+
+type Validator interface {
+	Validate(str string) error
+}
+
+type Policies struct {
+	minCharacters            int
+	minLowerCaseCharacters   int
+	minUpperCaseCharacters   int
+	minDigits                int
+	minSpecialCharacters     int
+	allowedSpecialCharacters string
+	digits                   *regexp.Regexp
+	specialCharacters        *regexp.Regexp
+}
+
+func NewPasswordPolicies(minCharacters, minLowerCaseCharacters, minUpperCaseCharacters, minDigits, minSpecialCharacters int,
+	allowedSpecialCharacters string) (Validator, error) {
+
+	digits := regexp.MustCompile("[0-9]")
+	if len(allowedSpecialCharacters) > 0 {
+		defaultSpecialCharacters = allowedSpecialCharacters
+	}
+	specialCharacters, err := regexp.Compile("[" + regexp.QuoteMeta(defaultSpecialCharacters) + "]")
+	if err != nil {
+		return nil, err
+	}
+	return &Policies{
+		minCharacters:            minCharacters,
+		minLowerCaseCharacters:   minLowerCaseCharacters,
+		minUpperCaseCharacters:   minUpperCaseCharacters,
+		minDigits:                minDigits,
+		minSpecialCharacters:     minSpecialCharacters,
+		allowedSpecialCharacters: allowedSpecialCharacters,
+		digits:                   digits,
+		specialCharacters:        specialCharacters,
+	}, nil
+}
+
+func NewDefaultPasswordPolicies() (Validator, error) {
+	return NewPasswordPolicies(0, 0, 0, 0, 0, "")
+}
```

Comment:
Do we really need this? If so it should hold sane defaults and not `0` everywhere. But we could also remove it and rely on the caller to pass correct values.